src/server/pps/server/pipeline_controller.go

```diff
@@ -190,14 +190,17 @@ func (pc *pipelineController) step(timestamp time.Time) (isDelete bool, retErr e
 		return true, nil
 	}
 	var stepErr stepError
+	// TODO(msteffen) should this fail the pipeline? (currently getRC will restart
+	// the pipeline indefinitely)
+	rc, restart, err := pc.getRC(ctx, pi)
+	if restart != "" {
+		return false, pc.restartPipeline(ctx, pi, rc, restart)
+	}
+	if err != nil && !errors.Is(err, errRCNotFound) {
```

Comment:
maybe `errRCNotFound` just shouldn't be returned. Also seems like `errUnexpectedRC` can be deleted so it's just k8s errors that cause this